src/test/java/com/wikia/webdriver/elements/mercury/components/discussions/common/category/CategoriesFieldset.java

```diff
@@ -0,0 +1,189 @@
+
+  @FindBy(className = "discussion-category-all")
+  private WebElement editableCategoryAll;
+
+  @FindBy(css = ".discussion-categories-list .sortable-item")
+  private List<WebElement> editableCategories;
+
+  @FindBy(className = "discussion-categories-edit-add-link")
+  private WebElement addCategoryLink;
+
+  @FindBy(css = ".discussion-categories-edit button.submit")
+  private WebElement approveChangesButton;
+
+  @FindBy(className = "categories-edit-info-message")
+  private WebElement infoMessage;
+
+  public boolean canEdit() {
+    WebElement svg = fieldset.findElement(By.tagName("use"));
+    return svg.getAttribute("xlink:href").equals("#pencil");
+  }
+
+  @CheckForNull
+  public CategoryPill findCategoryWith(final String categoryName) {
+    CategoryPill result = null;
+
+    for (int i = 0; i < categories.size() && result == null; i++) {
+      WebElement category = categories.get(i);
+      if (category.getText().equals(categoryName)) {
+        result = new CategoryPill(category, i);
+      }
+    }
+
+    return result;
+  }
+
+  /**
+   * @param position - category position, first category is 0 - All
+   * @return category pill
+   * @throws IllegalArgumentException when category is not found
+   */
+  public CategoryPill clickCategoryAt(final int position) {
+    return withBoundaryCheck(categories, position, webElement -> {
+      final CategoryPill result = new CategoryPill(webElement, position);
+      webElement.click();
+      return result;
+    });
+  }
+
+  private <T> T withBoundaryCheck(final List<WebElement> elements, final int position, Function<WebElement, T> transform) {
+    final int size = elements.size();
+
+    if (size > position) {
+      return transform.apply(elements.get(position));
+    } else {
+      throw new IllegalArgumentException("You wanted to click category on position " + position + " but there is only " + size + " categories.");
+    }
+  }
+
+  public CategoriesFieldset clickEdit() {
+    editLink.click();
+    return this;
+  }
+
+  /**
+   * For mobile "All" category is not visible, that is why it's always false - can not be edited.
+   *
+   * @return true if "All" category can be edited
+   */
+  public boolean canEditAllCategory() {
+    return isMobile() ? false
+        // please remember to change this line when SOC-3793 is done - "isCategoryEditable(editableCategoryAll)"
+        : !editableCategoryAll.findElement(By.className("fancy-checkbox-span")).getAttribute("class").contains("disabled");
+  }
+
+  private boolean isMobile() {
+    return !label.isDisplayed();
+  }
+
+  public boolean canEditGeneralCategory() {
+    return isCategoryEditable(findEditableCategoryWith(GENERAL_CATEGORY_NAME));
+  }
+
+  private WebElement findEditableCategoryWith(final String categoryName) {
+    WebElement result = null;
+
+    for (WebElement element : editableCategories) {
+      final String value = element.findElement(By.cssSelector("input[type='text']")).getAttribute("value");
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "input[type='text']" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1192)